packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1230,6 +1230,47 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     });
   }
 
+  /// Whether this has been disposed.
+  ///
+  /// If asserts are disabled, this property is always null.
+  bool? get debugDisposed {
+    bool? disposed;
+    assert(() {
+      disposed = _debugDisposed;
+      return true;
+    }());
+    return disposed;
+  }
+
+  bool _debugDisposed = false;
+
+  /// Release any resources held by this render object.
+  ///
+  /// If this render object has created any children directly, it should dispose
+  /// of those children in this method as well. However, it should not
+  /// dispose of children that were created by some other object, such as
+  /// a [RenderObjectElement].
+  ///
+  /// If [isRepaintBoundary] returns true, the layer tree rooted at this
+  /// object's layer will also dispose.
+  @mustCallSuper
+  void dispose() {
+    assert(!_debugDisposed);
+    if (isRepaintBoundary) {
+      _layer = null;
+    }
+    assert(() {
+      visitChildren((RenderObject child) {
+        assert(child._debugDisposed);
```

Comment:
nit: access via regular getter?

Reply:
Done - I was avoiding the `!` but it doesn't really matter much for asserts and will be less confusing if someone modifies the getter